styx-scheduler-service/src/main/java/com/spotify/styx/state/QueuedStateManager.java

```diff
@@ -200,13 +192,11 @@ private void initialize(WorkflowInstance workflowInstance) {
         }
 
         // Write new state to datastore (or remove it if terminal)
-        final long nextCounter = persistentState.get().counter() + 1;
+        final long nextCounter = currentRunState.get().counter() + 1;
         if (nextRunState.state().isTerminal()) {
           tx.deleteActiveState(event.workflowInstance());
         } else {
-          final PersistentWorkflowInstanceState nextPersistentState =
-              PersistentWorkflowInstanceState.of(nextRunState, nextCounter);
-          tx.updateActiveState(event.workflowInstance(), nextPersistentState);
+          tx.updateActiveState(event.workflowInstance(), nextRunState);
```

Comment:
`nextCounter` is lost here, should be set in `nextRunState`

Reply:
I actually moved the counter increasing within the `transition` method in `RunState`, so it's done "automatically".